### ISSUE TITLE
FolderContentsData not converting to JSON

### ISSUE THREAD

**Issue by tylerwarner33** [autodesk-platform-services/aps-sdk-net#107]
The Autodesk.DataManagement.Model.FolderContents result is returning correctly when I use this method:

``` csharp
public async Task<FolderContents> GetFolderContentsDataByIdAsync(string projectId, string folderId, List<FilterType> filterType = null)
{
    ThreeLeggedToken token = await _accThreeLeggedTokensService.GetThreeLeggedTokenAsync();
    FolderContents folderContents = await APSClientHelper.DataManagementClient.GetFolderContentsAsync(
        projectId: projectId,
        folderId: folderId,
        accessToken: token.AccessToken,
        filterType: filterType);

    return folderContents;
}
```

As shown here:
![image](https://github.com/user-attachments/assets/a5fb2a5b-635d-4e82-ac41-7c20508cf3b5)
![image](https://github.com/user-attachments/assets/3b79b764-ae75-446c-9de6-9873c9621b74)

However, FolderContents.Data (FolderContentsData) is not converting to JSON & therefore is not showing up in the response.

![image](https://github.com/user-attachments/assets/6d25d094-5290-4a19-8ef3-5181af5a70f3)

This was working in this previous [version](https://github.com/autodesk-platform-services/aps-sdk-net/blob/ae3f1eaf517f63e1614ef3f58733f6215fbf90a4/datamanagement/source/Model/FolderContentsData.gen.cs) vs the current [version](https://github.com/autodesk-platform-services/aps-sdk-net/blob/main/datamanagement/source/Model/FolderContentsData.gen.cs).

Nuget package versions currently being used:

``` xml
<PackageReference Include="Autodesk.Authentication" Version="2.0.0-beta4" />
<PackageReference Include="Autodesk.Construction.AccountAdmin" Version="2.0.0-beta" />
<PackageReference Include="Autodesk.DataManagement" Version="2.0.0-beta4" />
<PackageReference Include="Autodesk.Oss" Version="2.0.0-beta2" />
```

**Comment by Arrotech** [comment]
Hi @tylerwarner33 ,

Look at the following snippet of code, how to cast the FolderData or ItemData. Here is the link to the [sample](https://github.com/autodesk-platform-services/aps-sdk-net/blob/main/samples/datamanagement.cs) take a look at **GetFolderContentsAsync** method

Let me know if this helps

`    public async Task GetFolderContentsAsync()
    {
        List<FilterType> filter_type = new List<FilterType> { FilterType.Items, FilterType.Folders };

        FolderContents folderContents = await dataManagementClient.GetFolderContentsAsync(projectId: project_id, folderId: folder_id, filterType: filter_type);
        List<IFolderContentsData> folderContentsData = folderContents.Data;
        foreach (var current in folderContentsData)
        {
            if (current is FolderData folder)
            {
                Console.WriteLine(folder.Id);
                Console.WriteLine(folder.Type);
                Console.WriteLine(folder.Relationships.Parent.Data.Type);
            }
            else if (current is ItemData item)
            {
                Console.WriteLine(item.Id);
                Console.WriteLine(item.Type);
            }
        }
    }`

**Comment by tylerwarner33** [comment]
Hello @Arrotech, thank you, I did see that sample. It is helpful for being able to cast & use the data within the code.

But, is there a way for the data from the http response (image 1, includes data) to be passed to the JSON response (image 2, doesn't include data))?

![image](https://github.com/user-attachments/assets/3b79b764-ae75-446c-9de6-9873c9621b74)
![image](https://github.com/user-attachments/assets/6d25d094-5290-4a19-8ef3-5181af5a70f3)

**Comment by Arrotech** [comment]
Hi @tylerwarner33 ,

At the moment I would say, no. You have to cast as **FolderData** or **ItemData** to access their data. This issue is noted and will be fixed.

Let me know what is not working for you (the data you can't access) so that we can help.

Thanks for reporting.

**Comment by sajith-subramanian** [comment]
Hi @tylerwarner33 - not sure I follow the issue completely. In the previous version, we had an issue where Foldercontents.Data would not return ItemData correctly. Since FolderContents can contain both FolderData & ItemData, in the new version we are now returning an Interface which needs to be cast and their data accessed.  Let me know if I am missing something.

**Comment by tylerwarner33** [comment]
Hello @sajith-subramanian, maybe the last version wasn't working exactly right with both the FolderData & ItemData.

The issue I'm having with this version is when the FolderContents object is returned via an endpoint (a controller response in this case), the IFolderContentsData is not converting to JSON. So the "data" object in the JSON (images shown in the original post) is showing the correct number of items (FolderData & ItemData) in the array, but they are all empty [ {}, {}, {}, {}, {}, {}, {} ].

All of the data is there in the code, but not in the controller response (images shown in the original post).

The [FolderContents](https://github.com/autodesk-platform-services/aps-sdk-net/blob/e9b599be00f6d38157adb0b8a48b060b07513494/datamanagement/source/Model/FolderContents.gen.cs#L40) class is using this data annotation ```[JsonConverter(typeof(FolderContentsDataConverter))]``` for the conversion which is coming from this [custom converter class](https://github.com/autodesk-platform-services/aps-sdk-net/blob/e9b599be00f6d38157adb0b8a48b060b07513494/datamanagement/source/custom-code/utils.cs#L10).

Let me know if you are getting a different result.

**Comment by sajith-subramanian** [comment]
Thanks for the details @tylerwarner33 . We will log this for further investigation.

**Comment by sajith-subramanian** [comment]
Fixed and is part of the latest release V2.0.0.
